File: trunk-zt/whale-index-spark/src/main/scala/com/odianyun/search/whale/index/scala/realtime/converter/IDConverter.scala
package com.odianyun.search.whale.index.scala.realtime.converter

import com.odianyun.search.whale.index.scala.common.{IndexInfo, ProcessContext, UpdatedIds}

/**
  * Created by cuikai on 16/7/5.
  */
trait IDConverter extends Serializable {

  def convert(indexInfo: IndexInfo, context: ProcessContext, ids: List[Long]): UpdatedIds
}

File: trunk-zt/whale-index/src/main/java/com/odianyun/search/whale/index/convert/IDConverter.java
package com.odianyun.search.whale.index.convert;

import java.util.List;

import com.odianyun.search.whale.index.api.common.UpdateType;

public interface IDConverter {
	
	List<Long> convert(List<Long> ids,int companyId) throws Exception;
}
